### PR TITLE
TaskCenter all the things

### PR DESCRIPTION
**Pull request by AhmedSoliman**
TaskCenter all the things

This migrates _most_ of the long-running tasks to task center, and we get to harvest the fruit of this investment.
Note that this is still not 100% of the migration, but it's a sufficient start to witness the value.

A good follow-up would be to adopt TaskCenter's potential of scheduling tasks on multiple runtimes and move the rocksdb
single-threaded writer runtime to be managed by TaskCenter as well, but that's for another time.

---
Stack created with [Sapling](https://sapling-scm.com). Best reviewed with [ReviewStack](https://reviewstack.dev/restatedev/restate/pull/1194).
* #1196
* #1195
* __->__ #1194
* #1181

### CODE REVIEW

File: crates/bifrost/src/service.rs [hunk 1]
```diff
@@ -44,19 +44,25 @@ impl BifrostService {
...
-        let join_handle = tokio::spawn({
-            async {
-                self.watchdog.run(drain).await?;
-                info!("Bifrost watchdog shutdown complete");
-                Ok(())
-            }
-        });
-        Ok(join_handle)
+        task_center().spawn(
+            TaskKind::BifrostBackgroundHiPri,
```
Comment (tillrohrmann) [inline crates/bifrost/src/service.rs:59]: Should we spell "HighPriority" out?

File: crates/worker/src/lib.rs [hunk 2]
```diff
@@ -416,92 +377,97 @@ impl Worker {
...
+        )?;
 
         tokio::select! {
             _ = shutdown => {
                 debug!("Initiating shutdown of worker");
 
-                // first we shut down the network which shuts down the consensus which shuts
-                // down the partition processors transitively
+                // This will only shutdown rocksdb writer thread. Everything else will respond to
+                // the cancellation signal idependently.
```
Comment (tillrohrmann) [inline crates/worker/src/lib.rs:469]: typo: independently

File: crates/worker/src/lib.rs [hunk 3]
```diff
@@ -416,92 +377,97 @@ impl Worker {
...
+            self.storage_query_postgres.run(),
+        )?;
+
+        // Kafka Ingress
+        tc.spawn(
+            TaskKind::SystemService,
+            "kafka-ingress",
+            None,
+            self.ingress_kafka.run(),
+        )?;
```
Comment (tillrohrmann) [inline crates/worker/src/lib.rs:436]: Why is Kafka not spawned as a child?

**Review by tillrohrmann — APPROVED** [review]
Really nice how the `TaskCenter` simplifies the task management in the different components :-) LGTM. +1 for merging.

File: crates/worker/src/lib.rs (hunk 3 above)
Comment (AhmedSoliman) [reply crates/worker/src/lib.rs:436]: oversight.

**Pull request merged**